### ISSUE TITLE
Discovery phase fails when test has nested class

### ISSUE THREAD

**Issue by 3flex** [JetBrains/spek#642]
This below test from detekt now fails during discovery phase after upgrading from 2.0.0 to 2.0.1. The check on line 7 here caused the failure: https://github.com/spekframework/spek/blob/2be7ff1e3e05f630fef2e5e92fce44dfd4503e9c/spek-runtime/src/commonMain/kotlin/org/spekframework/spek2/runtime/util/ClassUtil.kt#L5-L9

This is because a nested class `NotConfiguredEmptyRule` is declared within the test, and that class does not have a qualified name

I cannot change it to an anonymous object in the test as the simple name will then have `$` characters in it, and these simple class names are validated internally by detekt which causes a different test failure.

```kotlin
class CommentInsideBlockSpec : Spek({

    class NotConfiguredEmptyRule : EmptyRule(Config.empty) {
        override fun visitBlockExpression(expression: KtBlockExpression) =
                expression.addFindingIfBlockExprIsEmpty()
    }

    val subject = NotConfiguredEmptyRule()

    describe("NotConfiguredEmptyRule rule") {

        it("does not find comment inside an empty block") {
            val findings = subject.lint("{/*comment*/}")
            assertThat(findings).isEmpty()
        }

        it("finds comments inside block") {
            val findings = subject.lint("{}")
            assertThat(findings).hasSize(1)
        }
    }
})
```

**Comment by raniejade** [comment]
Weird, not sure why `NotConfiguredEmptyRule` is being passed to `extractPackageAndClassNames` since it does not inherit `Spek` (filtered [here](https://github.com/spekframework/spek/blob/2.x/spek-runtime/src/jvmMain/kotlin/org/spekframework/spek2/runtime/JvmDiscoveryContextFactory.kt#L53)). I have to investigate it further but in the meantime you can annotate `NotConfiguredEmptyRule` with `@org.spekframework.spek2.meta.Ignore`.

**Comment by raniejade** [comment]
Resolved by #650
